napari/layers/text/text.py

```diff
@@ -0,0 +1,820 @@
+                selected.append(ind)
+        self._selected_view = selected
+        if len(selected) == 0:
+            self.selected_data
+        self._selected_box = self.interaction_box(self._selected_view)
+
+        self._set_highlight(force=True)
+        self._update_thumbnail()
+        self._update_coordinates()
+        self.events.set_data()
+
+    def _set_highlight(self, force=False):
+        """Render highlights of text including boundaries, vertices,
+        interaction boxes, and the drag selection box when appropriate
+
+        Parameters
+        ----------
+        force : bool
+            Bool that forces a redraw to occur when `True`
+        """
+        # Check if any point ids have changed since last call
+        if (
+            self.selected_data == self._selected_data_stored
+            and self._value == self._value_stored
+            and np.all(self._drag_box == self._drag_box_stored)
+        ) and not force:
+            return
+        self._selected_data_stored = copy(self.selected_data)
+        self._value_stored = copy(self._value)
+        self._drag_box_stored = copy(self._drag_box)
+
+        if self._mode == Mode.SELECT and (
+            self._value is not None or len(self._selected_view) > 0
+        ):
+            if len(self._selected_view) > 0:
+                index = copy(self._selected_view)
+                if self._value is not None:
+                    hover_point = list(self._indices_view).index(self._value)
+                    if hover_point in index:
+                        pass
+                    else:
+                        index.append(hover_point)
+                index.sort()
+            else:
+                hover_point = list(self._indices_view).index(self._value)
+                index = [hover_point]
+
+            self._highlight_index = index
+        else:
+            self._highlight_index = []
+
+        pos = self._selected_box
+        if pos is None and not self._is_selecting:
+            pos = np.zeros((0, 2))
+        elif self._is_selecting:
+            pos = create_box(self._drag_box)
+            pos = pos[list(range(4)) + [0]]
+        else:
+            pos = pos[list(range(4)) + [0]]
+
+        self._highlight_box = pos
+        self.events.highlight()
+
+    def _update_thumbnail(self):
+        """Update thumbnail with current points and colors."""
+        colormapped = np.zeros(self._thumbnail_shape)
+        colormapped[..., 3] = 1
+
+        colormapped[..., 3] *= self.opacity
+        self.thumbnail = colormapped
+
+    def add(self, coord):
+        """Adds text at coordinate.
+
+        Parameters
+        ----------
+        coord : sequence of indices to add point at
+        """
+        coords = np.append(self.text_coords, [coord], axis=0)
+        text = self.text
+
+        # Set the new text to 'EditMe' if none has been specified
+        new_text = self.new_text
+        if self.new_text == '':
+            new_text = 'EditMe'
```

Comment:
Not quite clear to me when you see `EditMe` vs what is already inside the text window. I'd also seems to save state in a way I don't understand. I'd like the newly created next to be whatever is currently displayed inside the textbox. It's possible for this not to be the case if you are just selecting text and then move to the text box.

Also right now if you just have a bunch of space like `  ` then it seems to behave a little strange.

Finally if we need a default - I prefer `edit` over `EditMe`

Reply:
My intent here was to make new text items have some text value if the user hasn't entered anything into the text box. If there is anything in the text box, it should use that value for the new text element. The rationale is that it's really difficult to select text that is empty for future editing. Does this make sense?

Can you clarify what the issues you're seeing where state doesn't seem to be preserved correctly? I am probably missing a call to an update function somewhere.

I can change `EditMe` to `edit`!

Reply:
I have updated the behavior such that the last selected text is preserved. For example, if you select an element with text "hi" and then switch to adding mode, the text of the element you add will also be "hi". This is also preserved if you switch to pan/zoom and then back to add. I think this is what you were describing. Please let me know what you think!